### PR TITLE
Minor refactor on the structure of the dataformat WRT M:M data

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Changed the "manytomany" key to "manyrelated" and mostly fleshed out that sub-dict.  Plus there are changes related to the serum tracer enrichment fraction method.

## Affected Issue Numbers

- Partially addresses  #484

## Code Review Notes

Most of these changes were made by a search & replace script of mine.  Very benign changes.  Just rearranged things to make more sense when applied to other many-related models.

## Checklist

- [ ] All issue requirements satisfied (or no linked issues)
  - Other components of the refactor will be addressed in a later PR 
- [ ] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
  - [x] Linting related to this change passes. Other linting issues (new to the main branch due to superlinter changes) are handled in other PRs
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [x] New test classes/functions tagged with `@tag("multi_working")`
- [x] [All (ci-run) tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] Test function tag `@tag("multi_working")` added and test class/function tags `@tag("multi_broken")`, `@tag("multi_unknown")`, or `@tag("multi_mixed")` removed or changed.

### CODE REVIEW

**Comment by hepcat72** [comment]
The pylint errors are fixed in a separate PR.

**Comment by hepcat72** [comment]
Implemented suggested changes from another PR here. Remote tests run much faster than my local runs, so going into draft to work on the errors.

**Comment by lparsons** [comment]
> Implemented suggested changes from another PR here. Remote tests run much faster than my local runs, so going into draft to work on the errors.

Yeah, I highly recommend letting tests run on GitHub instead of locally. I usually create branches and only create a PR once the tests are passing.

**Comment by hepcat72** [comment]
> > Implemented suggested changes from another PR here. Remote tests run much faster than my local runs, so going into draft to work on the errors.
> 
> Yeah, I highly recommend letting tests run on GitHub instead of locally. I usually create branches and only create a PR once the tests are passing.

Well, I hadn't decided where I was going to implement the PR review items from the peak group label PR. In fact, I'd forgotten about them. Decided to implement them here after I'd already created the PR.

I usually run select tests I know I need to get to work locally. And I don't like to check in changes until they work. Usually, that's enough. But in some cases, I'm not 100% sure any other tests are broken or not. Currently, it takes 15m to run tests on github and much longer locally. So if I ever feel like I need to run all, I check in and push.

This was a special case.

File: DataRepo/models/animal_label.py [hunk 1]
```diff
@@ -83,9 +83,15 @@ def serum_tracers_enrichment_fraction(self):
...
-            final_serum_tracer_peak_group_elem = final_serum_tracer_peak_groups.get(
+            final_serum_tracer_peak_groups_elem = final_serum_tracer_peak_groups.filter(
                 peak_group_labels__element__exact=self.element,
             )
+            # tracer_compounds has only compounds with this element and this assures all necessary peak groups have the
+            # labeled element among its peak group label records
+            if final_serum_tracer_peak_groups_elem.count() != len(tracer_compounds):
+                raise MissingPeakGroupLabel(
+                    final_serum_tracer_peak_groups, self.element
+                )
```
Comment (hepcat72) [inline DataRepo/models/animal_label.py:94]: C. This and the below changes fixed a mistake that I'm surprised didn't make tests fail in the animal label PR. But it definitely popped up an exception when I ran the server and did a search. I had mistakenly implemented one element = one peak group.  But going down to one element corresponds to multiple tracer compounds with that element. This change fixes that.

**Comment by hepcat72** [comment]
Most of the changes are mundane, except the model file changes.

**Comment by hepcat72** [comment]
Man, I keep misstepping with this test failure. When I pushed before ending the workday yesterday, the tests I'd run (which succeeded) didn't include the newly re-named test, which of course failed.  Fixing it now.  This should finally be ready for review shortly.

**Pull request merged**